src/pages/signin/SAMLSignInPage/index.native.js

```diff
@@ -19,39 +26,61 @@ const defaultProps = {
     credentials: {},
 };
 
-const renderLoading = () => <FullScreenLoadingIndicator />;
+// const renderLoading = () => <FullScreenLoadingIndicator />;
 
 function SAMLSignInPage({credentials}) {
     const webViewRef = useRef();
     const samlLoginURL = `${CONFIG.EXPENSIFY.SAML_URL}?email=${credentials.login}&referer=${CONFIG.EXPENSIFY.EXPENSIFY_CASH_REFERER}`;
-
     /**
-    * Handles in-app navigation once we get a response back from Expensify
-    *
-    * @param {String} params.type
-    * @param {String} params.url
-    */
+     * Handles in-app navigation once we get a response back from Expensify
+     *
+     * @param {String} params.type
+     * @param {String} params.url
+     */
     const handleNavigationStateChange = useCallback(
         ({type, url}) => {
             const searchParams = new URLSearchParams(new URL(url).search);
             if (searchParams.has('shortLivedAuthToken')) {
                 const shortLivedAuthToken = searchParams.get('shortLivedAuthToken');
                 Session.signInWithShortLivedAuthToken(credentials.login, shortLivedAuthToken);
-                return;
+                Navigation.navigate(ROUTES.HOME);
+            }
+            if (searchParams.has('error')) {
+                // Run the Onyx action to set an error state on the sign in page
+                // Currently this is what's going to trigger because the backend isn't redirecting SAML correctly
+                Navigation.navigate(ROUTES.HOME);
             }
         },
-        [webViewRef],
+        [credentials.login],
     );
     return (
-        <WebView
-            ref={webViewRef}
-            originWhitelist={['https://*']}
-            source={{uri: samlLoginURL}}
-            incognito // 'incognito' prop required for Android, issue here https://github.com/react-native-webview/react-native-webview/issues/1352
-            startInLoadingState
-            renderLoading={renderLoading}
-            onNavigationStateChange={handleNavigationStateChange}
-        />
+        <ScreenWrapper
+            shouldShowOfflineIndicator={false}
+            includeSafeAreaPaddingBottom={false}
```

Comment:
We'll need to add a component display name and then pass it here 

```suggestion
            includeSafeAreaPaddingBottom={false}
            testID={SAMLSignInPage.displayName}
```